### PR TITLE
feat(common): introduce AuxiliaryStorageWrap to remove boilerplate code

### PR DESCRIPTION
**Pull request by playX18**

### CODE REVIEW

File: common/src/auxiliary/mod.rs [hunk 1]
```diff
@@ -136,3 +140,118 @@ impl<K1, K2, V> Default for DoubleBTreeMap<K1, K2, V> {
...
+    fn contains_keys(key1: &Self::Key1, key2: &Self::Key2) -> bool {
+        T::with_storage_mut(|map| map.contains_keys(key1, key2))
+    }
+
+    fn mutate<R, F: FnOnce(&mut Option<Self::Value>) -> R>(
+        _key1: Self::Key1,
+        _key2: Self::Key2,
+        _f: F,
+    ) -> R {
+        unimplemented!()
```
Comment (breathx) [inline common/src/auxiliary/mod.rs:195]: shouldn't these be implemented already?

File: common/src/auxiliary/mod.rs [hunk 2]
```diff
@@ -136,3 +140,118 @@ impl<K1, K2, V> Default for DoubleBTreeMap<K1, K2, V> {
         Self::new()
     }
 }
+
+/// An auxiliary storage wrapper type.
+///
+/// Implements DoubleMapStorage and traits like [`IterableByKeyMap`] for such type automatically.
+pub trait AuxiliaryStorageWrap {
+    type Key1: Ord + Clone;
```
Comment (breathx) [inline common/src/auxiliary/mod.rs:148]: storage wrap could be just single map

Comment (breathx) [reply common/src/auxiliary/mod.rs:148]: so assocs here seem incorrect to me

**Review by breathx — CHANGES_REQUESTED** [review]

File: common/src/auxiliary/mod.rs (hunk 2 above)
Comment (breathx) [reply common/src/auxiliary/mod.rs:148]: it should be just across T, where T is stored in thread local

**Review by breathx — APPROVED** [review]

File: common/src/auxiliary/mod.rs (hunk 2 above)
Comment (playX18) [reply common/src/auxiliary/mod.rs:148]: resolved in DM, we'll settle on using two traits for different storages

**Pull request merged**